draft-ietf-httpbis-sfbis.md

```diff
@@ -1102,37 +1102,31 @@ key           = ( lcalpha / "*" )
 lcalpha       = %x61-7A ; a-z
 param-value   = bare-item
 
-sf-dictionary  = dict-member *( OWS "," OWS dict-member )
-dict-member    = member-key ( parameters / ( "=" member-value ))
-member-key     = key
-member-value   = sf-item / inner-list
+sf-dictionary = dict-member *( OWS "," OWS dict-member )
+dict-member   = member-key ( parameters / ( "=" member-value ))
+member-key    = key
+member-value  = sf-item / inner-list
 
 sf-item   = bare-item parameters
 bare-item = sf-integer / sf-decimal / sf-string / sf-token
-            / sf-binary / sf-boolean / sf-date
+            / sf-binary / sf-boolean / sf-date / sf-displaystring
 
-sf-integer = ["-"] 1*15DIGIT
+sf-integer       = ["-"] 1*15DIGIT
+sf-decimal       = ["-"] 1*12DIGIT "." 1*3DIGIT
+sf-string        = DQUOTE *( unescaped / "%" / bs-escaped ) DQUOTE
+sf-token         = ( ALPHA / "*" ) *( tchar / ":" / "/" )
+sf-binary        = ":" *base64 ":"
+sf-boolean       = "?" ( "0" / "1" )
+sf-date          = "@" sf-integer
+sf-displaystring = "%" DQUOTE *( unescaped / "\" / pct-encoded ) DQUOTE
 
-sf-decimal  = ["-"] 1*12DIGIT "." 1*3DIGIT
+base64       = ALPHA / DIGIT / "+" / "/" / "="
```

Comment:
base64 is really:

```abnf
base64 = *(ALPHA / DIGIT / "+" / "/") *"="
```

Reply:
Yeah, I was wondering how far to go down that rabbit hole. I think that's a nice balance.

Reply:
Yup.  `*3"="` is one step too many perhaps.

Reply:
In particular, I don't want to imply that we're validating base64 here -- same as the percent encoding.